### PR TITLE
adding demo testing tests

### LINKED ISSUE #25: Add unit test and output verification gh actions for the demo

### PR DESCRIPTION
**Pull request by shawn-hurley**
fixes #25

### CODE REVIEW

**Review by pranavgaikwad — APPROVED** [review]
Added minor questions, overall looks good. Thanks @shawn-hurley

File: rule-example.yaml [hunk 1]
```diff
@@ -1,54 +1,43 @@
 ---
 - message: all go files
+  ruleID: file-001
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply rule-example.yaml:3]: Probably, we should add a check during rule parsing such that this is not allowed. I can add a follow up issue thoughts?

File: provider/provider.go [hunk 2]
```diff
@@ -50,10 +56,34 @@ func (p *ProviderCondition) Evaluate(log logr.Logger, ctx map[string]interface{}
...
+	for _, inc := range resp.Incidents {
+		links := []engine.ExternalLinks{}
+		for _, link := range inc.Links {
+			links = append(links, engine.ExternalLinks{
+				URL:   link.URL,
+				Title: link.Title,
+			})
+		}
+		// Default to -1 if no effort is assinged.
+		// TODO(shawn-hurley evaluate if this is the correct default or if we want the end facing doc to have nil as an option.)
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply provider/provider.go:69]: Yeah, -1 feels like a hack for not set. Want to add it to the API discussions that we need to have when Fabian is back?

[the remark this answers is not in the document]
Comment (shawn-hurley) [reply provider/provider.go:69]: https://github.com/konveyor/analyzer-lsp/issues/28#issuecomment-1301132323 Adding comment here. I think we have time to fix this after merge :)

File: rule-example.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply rule-example.yaml:3]: https://github.com/konveyor/analyzer-lsp/issues/33

**Pull request merged**